power_grid_model_c/power_grid_model/include/power_grid_model/component/transformer_tap_regulator.hpp

```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+#include "base.hpp"
+#include "regulator.hpp"
+
+#include "../auxiliary/input.hpp"
+#include "../auxiliary/output.hpp"
+#include "../auxiliary/update.hpp"
+#include "../calculation_parameters.hpp"
+#include "../common/common.hpp"
+
+namespace power_grid_model {
+
+class TransformerTapRegulator : public Regulator {
```

Comment:
this should be added to `all_components.hpp` to unblock the transformer ranking PRs